tests/cases/compiler/tupleTypes.ts

```diff
@@ -17,6 +17,13 @@ t = [1, "hello"];     // Ok
 t = ["hello", 1];     // Error
 t = [1, "hello", 2];  // Ok
 
+var et: [];
+var et0 = et[0];  // never
+var et0: never;
+
+et = [];    // Ok

```

Comment:
can you add this same check in a new file, something like tupleTypesStrictNullCheck.ts, with the first line `// @strictNullChecks: true` to turn on strict null checks?